### PR TITLE
Bug 1956208: openstack: add validation for volume types

### PR DESCRIPTION
**Pull request by EmilienM**
When an user wants to deploy with rootVolume, they need to specify
a volume type.
We already checked that the volume type was set but we didn't check that
the type was actually existing.

With this patch, we now check that the type exists.

If the Volume Type doesn't exist, it'll show this error message:
```
FATAL failed to fetch Metadata: failed to load asset "Install Config": compute[0].platform.openstack.rootVolume.type.type: Invalid value: "wrong": Volume Type either does not exist in this cloud, or is not available
```

Note: this patch also adds test coverage when the type is an empty string, which wasn't done before.

### CODE REVIEW

**Comment by pierreprinetti** [comment]
The failing CI tests are unrelated to this change. However, it seems that there are unit tests to fix

File: pkg/asset/installconfig/openstack/validation/cloudinfo.go [hunk 1]
```diff
@@ -195,6 +197,11 @@ func (ci *CloudInfo) collectInfo(ic *types.InstallConfig, opts *clientconfig.Cli
 		return err
 	}
 
+	ci.VolumeTypes, err = ci.getVolumeTypes()
```
Comment (mandre) [inline pkg/asset/installconfig/openstack/validation/cloudinfo.go:200]: Possibly, we only gather the volume types when using root volumes.

**Review by mandre — COMMENTED** [review]
/lgtm

**Comment by mandre** [comment]
/assign @staebler 
For approving the vendoring change.

File: pkg/asset/installconfig/openstack/validation/cloudinfo.go (hunk 1 above)
Comment (EmilienM) [reply pkg/asset/installconfig/openstack/validation/cloudinfo.go:200]: ack I'll do it later

**Comment by EmilienM** [comment]
my patch has nothing to do with e2e-libvirt, can we override it please?
